src/kaldi_recognizer.cc

```diff
@@ -408,6 +415,109 @@ bool KaldiRecognizer::GetSpkVector(Vector<BaseFloat> &out_xvector, int *num_spk_
     return true;
 }
 
+// Customized CompactLatticeToWordProns method in Kaldi to return weights
```

Comment:
Oof, that’s a big method with no comments at all. Could we add some basic comments that explain what the various blocks in this method are doing?

Reply:
Hmm yeah, I copied most of it from Kaldi which had no comments originally but I realize that has been a great disadvantage for me too to understand the code. I've added some high-level comments in places in the method.